projects/lib/src/board/result.h

```diff
@@ -22,9 +22,59 @@
 #include "side.h"
 #include <QMetaType>
 #include <QCoreApplication>
+#include <cmath>
 
 namespace Chess {
 
+/*!
+ * \brief The r-mobility (https://wiki.chessdom.org/R-Mobility) result of a chess game
+ *
+ * The RMobilityResult class is used to store the rmobility result of a chess game,
+ * and to compare it to other results.
+ */
+class LIB_EXPORT RMobility
+{
+	Q_DECLARE_TR_FUNCTIONS(RMobility)
+public:
+	explicit RMobility() : m_isValid(false), m_result(0.0f), m_winning(Side::White) {}
```

Comment:
Here also I'd like the implementations to be in the source file, and preferably separate header and source files for this class.